### PR TITLE
feat: new category table component

### PR DESCRIPTION
**Pull request by emilyjablonski**
# Pull Request Template

## Issue Overview

This PR addresses #2644

- [x] This change addresses the issue in full

## Description

This PR adds a new table component for use on DAHLIA that we do want to have in Bloom. You can check out the issue for screenshots, but it has passed an SF design PR review, so the review here is more code-based.

The diff is fairly large because in order to accomplish the AC, I needed to be able to pass cell-specific settings, so I wasn't able to just pass in string/JSX content for each cell as we were doing. I made a breaking change to the `StandardTable` component in terms of how we pass in individual cell content to do this. Now we are passing in an object `{content: "foo", optionalSetting: "bar"}`, rather than just `foo`, which is responsible for the diff. While a bigger code change, I think this is generally a good move forward for adding flexibility to the table content.

In terms of the PR-specific changed files:
* The new `CategoryTable` is very similar to our `StackedTable` component, so the `CategoryTable.tsx` file itself fairly small.
* I slightly refactored the StackedTable component for cleanliness and style needs, pulling inline styles into new classnames in the `tables.scss` file, and added the new `CategoryTable` class names there.
* The `StandardTable.tsx` file includes the breaking cell content change, and the new prop on the cell needed.
* I added typedoc comments on the prop interfaces into the table files , which get pulled directly into the Storybook docs page for each component.

## How Can This Be Tested/Reviewed?

Code review of the changes, and ensuring our existing tables work as they did before.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have added QA notes to the issue with applicable URLs
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [x] I have reviewed the changes in a mobile view
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [x] I have exported any new pieces added to ui-components
- [ ] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

**Comment by emilyjablonski** [comment]
TODO: Breaking change commit

**Review by seanmalbert — CHANGES_REQUESTED** [review]
@emilyjablonski ,

Can you add to the stories to have an example that uses `optionalSetting` for the `StandardTable` component.

**Comment by emilyjablonski** [comment]
@seanmalbert Happy to do so, but the CategoryTable stories use those settings, which under the hood uses the StandardTable, and we don't have a use case at the moment of a StandardTable with mobile replacements.

**Review by jaredcwhite — CHANGES_REQUESTED** [review]
@emilyjablonski This approach looks good to me. I didn't look closely at the diff for all the site pages but it should be fine. I noticed a couple small issues in the table HTML. Otherwise, LGTM!

**Review by YazeedLoonat — APPROVED** [review]
From convo with emily this looks good to me

File: ui-components/src/tables/StackedTable.tsx [hunk 1]
```diff
@@ -1,43 +1,54 @@
...
   headers: TableHeaders
+  /** Headers hidden on desktop views */
   headersHiddenDesktop?: string[]
+  /** The table data passed as records of column name to cell data */
   stackedData?: Record<string, StackedTableRow>[]
+  /** A class name applied to the root of the table */
   className?: string
 }
 
 const StackedTable = (props: StackedTableProps) => {
```
[the remark this answers is not in the document]
Comment (seanmalbert) [reply ui-components/src/tables/StackedTable.tsx:27]: Referenced in new issue https://github.com/bloom-housing/bloom/issues/2678

**Review by seanmalbert — APPROVED** [review]

**Review by jaredcwhite — APPROVED** [review]

**Pull request merged**